### PR TITLE
[tests-only] Added acceptance tests for allow certain groups to create link share

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
Acceptance tests have been added to the new feature at https://github.com/owncloud/core/pull/38980

## Related Issue

- Fixes https://github.com/owncloud/core/issues/39002

## How Has This Been Tested?

- 🤖 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by kiranparajuli589** [comment]
failed on php-style

**Comment by JanAckermann** [comment]
Beautiful! :rocket:

I would require one more scenarios:
Edge-Case if allow sharing only for certain groups is enabled but no group is set, creating shares should be possible

**Comment by kiranparajuli589** [comment]
> Beautiful! 
> 
> I would require one more scenario:
> Edge-Case if allow sharing only for certain groups is enabled but no group is set, creating shares should be possible
> 
> Multiple groups combinations could also be possible whether in `user` or `sharing only for certain groups setting` but this should be already covered by unit tests

I'll add them right away

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProxySmoke-8-2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31259/172/1

**Comment by kiranparajuli589** [comment]
>  Acceptance tests pipeline **apiProxySmoke-8-2-mariadb10.2-php7.4** failed. The build has been cancelled.
> 
> https://drone.owncloud.com/owncloud/core/31259/172/1

random failure
```console

curl: (6) Could not resolve host: proxy
--
34 | Server on http://proxy is down or not working correctly.
35 | make: *** [test-acceptance-api] Error 98

```

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingPublic2-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31260/151/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingPublic2-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31261/151/1

**Comment by JanAckermann** [comment]
Random crash again :/ restarted.

**Comment by JanAckermann** [comment]
Something mixed up, can you rebase ?

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingPublic2-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31269/151/1

**Review by AlexAndBear — APPROVED** [review]

File: tests/acceptance/features/apiSharePublicLink2/allowGroupToCreatePublicLinks.feature [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      | name        | shared-link       |
+    And parameter "public_share_sharers_groups_allowlist_enabled" of app "files_sharing" has been set to "yes"
+    And parameter "public_share_sharers_groups_allowlist" of app "files_sharing" has been set to '["grp1"]'
+    When user "Alice" deletes public link share named "shared-link" in file "fileToShare.txt" using the sharing API
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "100"
+    And as user "Alice" the file "fileToShare.txt" should not have any shares
+
+
+  Scenario: creating a new link share is not restricted if no groups are inside the allowed public share sharers groups even if allowlist is enabled
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharePublicLink2/allowGroupToCreatePublicLinks.feature:72]: This is an interesting scenario. If the admin enables this feature and specifies no groups in the "allow list" then what should happen? Should that mean that there are no users allowed to create public links? (Effectively public links are disabled) Or do we keep the behavior written in this scenario - setting `public_share_sharers_groups_allowlist_enabled` has no effect in this case.

@JanAckermann

Comment (JanAckermann) [reply tests/acceptance/features/apiSharePublicLink2/allowGroupToCreatePublicLinks.feature:72]: @phil-davis I decided myself if public_share_sharers_groups_allowlist_enabled but no groups are set, that there will be no public share restriction, to keep support cases as low as possible. 
("Hey we don't see the sharing tab anymore").

But for sure this needs to be addressed in PB2

**Pull request merged**